modules/internal/ChapelBase.chpl

```diff
@@ -1293,12 +1311,15 @@ module ChapelBase {
   }
 
   inline proc %=(ref lhs:int(?w), rhs:int(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0 then
+        halt("Attempt to compute a modulus by zero");
     __primitive("%=", lhs, rhs);
   }
   inline proc %=(ref lhs:uint(?w), rhs:uint(w)) {
-    __primitive("%=", lhs, rhs);
-  }
-  inline proc %=(ref lhs:real(?w), rhs:real(w)) {
```

Comment:
You've read the diff correctly. From the PR msg:

> Dropped `proc %=(ref lhs:real(?w), rhs:real(w))`, since `%` is not supported for real-types

Is this incorrect, or is there a separate reason to support `%=` for reals, when `%` is not supported?